Xamarin.Forms.Core/Internals/NameScope.cs

```diff
@@ -23,6 +24,15 @@ void INameScope.RegisterName(string name, object scopedElement)
 			_names[name] = scopedElement;
 		}
 
+		//used by VS Live Visual Tree
+		internal string NameOf(object scopedObject)
+		{
+			var index = _names.Values.IndexOf(scopedObject);
+			if (index < 0)
+				return null;
+			return _names.Keys.Skip(index - 1).FirstOrDefault();
```

Comment:
Does this work correctly if the Value is found at index `0` or `1`?

```c#
"abc".Skip(0 - 1).FirstOrDefault(); // => "a"
"abc".Skip(1 - 1).FirstOrDefault(); // => "a"
```

Reply:
I didn't meant the `-1`. thanks for spotting